sympy/combinatorics/fp_groups.py

```diff
@@ -597,9 +597,9 @@ def _check(F):
 
     h_gens, h_rels = _get_pres(parent_group, H)
     T = None
-    # If the generators of `G` is a proper subgroup of the
-    # generators of `parent_group` return the `FpGroup` on a new presentation.
-    if isinstance(G, list) and all(elem in parent_group.generators for elem in G):
+    # If `G` is generated by any set of elements of the
+    # `parent_group` return the `FpGroup` on a new presentation.
```

Comment:
Again, `G` is always generated by any set of elements so no need to say anything about it. Also, I think you meant `on a new free group`